src/types/utils.cpp

```diff
@@ -658,6 +659,24 @@ bool Utils::HandleWantsOverlappedIo(HANDLE handle) noexcept
     return status == 0 && WI_AreAllFlagsClear(modeInfo.Mode, FILE_SYNCHRONOUS_IO_ALERT | FILE_SYNCHRONOUS_IO_NONALERT);
 }
 
+Utils::Pipe Utils::CreatePipe(DWORD bufferSize)
+{
+    wil::unique_hfile rx, tx;
+    THROW_IF_WIN32_BOOL_FALSE(::CreatePipe(rx.addressof(), tx.addressof(), nullptr, bufferSize));
+    return { std::move(tx), std::move(rx) };
+}
+
+Utils::Pipe Utils::CreateOverlappedPipe(DWORD bufferSize)
+{
+    const auto rnd = til::gen_random<uint64_t>();
+    const auto name = fmt::format(FMT_COMPILE(LR"(\\.\pipe\{:016x})"), rnd);

```

Comment:
is this a limited namespace that will run out? will the security descriptor on this object be set such that it prevents tampering, or access by other processes on the system which did not inherit it?

Reply:
The latest commit changes this code to create proper anonymous pipes that are simultaneously overlapped. They don't show up in PipeList, Process Hacker, nor Process Explorer. I've also tested it on Windows 10 and it works perfectly fine. According to the internal code history, the NPFS implementation hasn't meaningfully changed in a long time so we should be good. (I've checked all the way back to RS1 and it hasn't changed since then. The file headers routinely point to them being written in the 90s.)

Reply:
The weirdest thing happened: The overlapped anonymous pipe (93930bb) worked absolutely perfect on my system yesterday, despite us using non-overlapped reads/writes in `ConptyConnection`. Now, a day later and after rebooting my system, it simply doesn't work anymore, unless I use overlapped reads/writes.